core/metamodel/src/main/java/org/apache/isis/core/metamodel/services/ServicesInjector.java

```diff
@@ -442,6 +456,17 @@ private static void addAssignableTo(final Class<?> type, final List<Object> cand
         .forEach(filteredServicesAndContainer::add);
     }
 
+    // -- LAZY INIT
+    
+    private Map<Class<?>, Object> initServiceByConcreteType(){
+        final Map<Class<?>, Object> map = _Maps.newHashMap();
+        for (Object service : services) {
+            final Class<?> concreteType = service.getClass();
+            map.put(concreteType, service);
+        }
+        return map;
+    }
+    
```

Comment:
that's the value supplier for the above _Lazy

Reply:
+1